### DISCUSSION TITLE
Clips Retain

### DISCUSSION THREAD

**Discussion by bm55** [blakeblackshear/frigate#512] [Ask A Question]
@blakeblackshear when does frigate delete the clips (only when drive is full or at certain time)? few days ago I removed and created new database, changed the config's global save_clips > retain to 1 day, also deleted the saved clips manually. But I see more than 2 days clips.

**Comment by blakeblackshear** [comment]
It should delete the files and the database entries every 10 minutes based on that config. It will only delete clips if the events are in the database. If you manually delete the database and it is recreated, the existing clips will stay indefinitely.

**Reply by bm55** [reply]
I deleted both clips and database then let Frigate create new db and clips. I can play 3 days clips in Media Browser but config is set to keep clips for only 1 day. Perhaps permission issue since the clips are stored on network drive - I will look into it.

**Reply by blakeblackshear** [reply]
Make sure you don't have any errors in the logs.

**Reply by bm55** [reply]
@blakeblackshear  there is no error, Beta 0.8.0 is working great. I also enabled the debug and see the delete database statements.
It looks like Frigate is deleting clips but it's not deleting all clips from previous day when I configure to keep the clips for 1 day only. I monitored the the number of clips from HA's media browser yesterday 1/13. It started adding clips on day 1/13 and also started deleting clips from 1/12 but 45 clips remains. Also, from 1/11 there is still 65 clips remaining.

![image](https://user-images.githubusercontent.com/7528001/104659025-c5d3fa00-5678-11eb-8e81-dbdf4d11ebdb.png)

> peewee                         DEBUG   : ('DELETE FROM "event" WHERE ((("event"."camera" = ?) AND ("event"."start_time" < ?)) AND ("event"."label" = ?))', ['garage', 1609645340.045119, 'dog'])


peewee                         DEBUG   : ('SELECT "t1"."id", "t1"."label", "t1"."camera", "t1"."start_time", "t1"."end_time", "t1"."top_score", "t1"."false_positive", "t1"."zones", "t1"."thumbnail" FROM "event" AS "t1" WHERE ((("t1"."camera" = ?) AND ("t1"."start_time" < ?)) AND ("t1"."label" = ?))', ['garage', 1610422940.048962, 'person'])


peewee                         DEBUG   : ('DELETE FROM "event" WHERE ((("event"."camera" = ?) AND ("event"."start_time" < ?)) AND ("event"."label" = ?))', ['garage', 1610422940.048962, 'person'])


peewee                         DEBUG   : ('SELECT "t1"."camera", "t1"."label", strftime(?, datetime("t1"."start_time", ?, ?)) AS "day", "t1"."zones", COUNT("t1"."id") AS "count" FROM "event" AS "t1" GROUP BY "t1"."camera", "t1"."label", strftime(?, datetime("t1"."start_time", ?, ?)), "t1"."zones"', ['%Y-%m-%d', 'unixepoch', 'localtime', '%Y-%m-%d', 'unixepoch', 'localtime'])
